tests/foreman/api/test_repository.py

```diff
@@ -519,204 +647,272 @@ def test_negative_create_label(self):
-            with self.subTest(content_type):
-                with self.assertRaises(HTTPError):
-                    entities.Repository(
-                        product=self.product,
-                        content_type=content_type,
-                        download_policy='on_demand',
-                    ).create()
+        with pytest.raises(HTTPError):
+            entities.Repository(**repo_options).create()
 
     @pytest.mark.tier1
-    def test_negative_create_checksum(self):
+    @pytest.mark.parametrize(
+        'repo_options',
+        **parametrized([{'checksum_type': gen_string('alpha'), 'download_policy': 'immediate'}]),
+        indirect=True,
+    )
+    def test_negative_create_checksum(self, repo_options):
         """Attempt to create repository with invalid checksum type.
 
         :id: c49a3c49-110d-4b74-ae14-5c9494a4541c
 
+        :parametrized: yes
+
         :expectedresults: A repository is not created and error is raised.
 
         :CaseImportance: Critical
         """
-        with self.assertRaises(HTTPError):
-            entities.Repository(
-                checksum_type=gen_string('alpha'), download_policy='immediate'
-            ).create()
+        with pytest.raises(HTTPError):
+            entities.Repository(**repo_options).create()
 
     @pytest.mark.tier1
-    def test_negative_create_checksum_with_on_demand_policy(self):
+    @pytest.mark.parametrize(
+        'repo_options',
+        **parametrized(
+            [
+                {'checksum_type': checksum_type, 'download_policy': 'on_demand'}
+                for checksum_type in ('sha1', 'sha256')
+            ]
+        ),
+        indirect=True,
+    )
+    def test_negative_create_checksum_with_on_demand_policy(self, repo_options):
         """Attempt to create repository with checksum and on_demand policy.
 
         :id: de8b157c-ed62-454b-94eb-22659ce1158e
 
+        :parametrized: yes
+
         :expectedresults: A repository is not created and error is raised.
 
         :CaseImportance: Critical
         """
-        for checksum_type in 'sha1', 'sha256':
-            with self.assertRaises(HTTPError):
-                entities.Repository(
-                    checksum_type=checksum_type, download_policy='on_demand'
-                ).create()
+        with pytest.raises(HTTPError):
+            entities.Repository(**repo_options).create()
 
     @pytest.mark.tier1
-    def test_negative_update_checksum_with_on_demand_policy(self):
+    @pytest.mark.parametrize(
+        'repo_options',
+        **parametrized(
+            [
+                {'checksum_type': checksum_type, 'download_policy': 'immediate'}
+                for checksum_type in ('sha1', 'sha256')
+            ]
+        ),
+        indirect=True,
+    )
+    def test_negative_update_checksum_with_on_demand_policy(self, repo):
         """Attempt to update the on_demand downloadpolicy on a already created repository with
```

Comment:
```suggestion
        """Attempt to update the on-demand download policy on a already created repository with
```
Just in case someone searches for this using plain text

Reply:
Fixed.